### PR TITLE
Tools: add IMU filter test tool

### PR DESCRIPTION
**Pull request by guglie**
Here's a tool to simulate IMU filtering to raw logs and see effects on simulated FFT graph.

More info here https://discuss.ardupilot.org/t/imu-filter-tool/43633

### CODE REVIEW

**Comment by Jaaaky** [comment]
Looks interesting, Thanks a lot.
Is it fully compatible with arduplane logs?

**Comment by guglie** [comment]
@Jaaaky 
I think it should support any vehicle type, but I've not tested on Arduplane logs, please tell me if it doesn't work on that so I can fix it.
For now it supports only the first IMU as it reads ACC1 and GYR1 log messages.

**Comment by khancyr** [comment]
very nice ! It seems good !

File: Tools/FilterTestTool/BiquadFilter.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
```
Comment (peterbarker) [inline Tools/FilterTestTool/BiquadFilter.py]: Do we really need python3?

Comment (amilcarlucas) [reply Tools/FilterTestTool/BiquadFilter.py]: python2 support is ending in 2019. It is a good idea to start using and testing as much python3 as possible.

Comment (amilcarlucas) [reply Tools/FilterTestTool/BiquadFilter.py]: @peterbarker The CI is failing, that is really strange.

**Comment by guglie** [comment]
Thank you @peterbarker for the review!
I've reformatted / cleaned the code and added a Readme

File: Tools/FilterTestTool/BiquadFilter.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+""" Ardupilot BiquadFilter
```
Comment (OXINARF) [inline Tools/FilterTestTool/BiquadFilter.py]: Please change all these to Ardu**P**ilot

Comment (guglie) [reply Tools/FilterTestTool/BiquadFilter.py]: Thanks Francisco, I forgot some of them

File: Tools/FilterTestTool/BiquadFilter.py (hunk 1 above)
Comment (guglie) [reply Tools/FilterTestTool/BiquadFilter.py]: @peterbarker I removed the `3` from the shebang line and made imports compatible with python2, but it is still working only under python3

**Comment by peterbarker** [comment]
Merged, thanks!

Getting it to work under Python2 would be nice, but merged there's a chance someone else will do it.

**Comment by guglie** [comment]
Thanks Peter!

**Comment by andyp1per** [comment]
Hey @guglie I get this on both python2 and python3:

andy@Eagle:~/github/ardupilot-master$ python3 ./Tools/FilterTestTool/run_filter_test.py
Loading 00000351.BIN...

LOG_BITMASK ,  931839.0
INS_GYRO_FILTER ,  100.0
INS_ACCEL_FILTER ,  20.0
INS_NOTCH_ENABLE ,  0.0
INS_LOG_BAT_OPT ,  2.0

32712 ACC logs for a duration of 32.7 secs
ACC frequency = 999.48 hz

32711 GYR logs for a duration of 32.7 secs
GYR frequency = 999.48 hz

INS_LOG_BAT_OPT was set to 2 which enables post filter logging,use pre filter logging to not sum multiple filter passes.
(set INS_LOG_BAT_OPT = 0)
Reading filter parameters to set initial filter values...
The firmware that produced this log does not support notch filter on accelerometer
Traceback (most recent call last):
  File "./Tools/FilterTestTool/run_filter_test.py", line 269, in <module>
    log_name, ACCEL_NOTCH_FILTER)
  File "/mnt/i/github/ardupilot-master/Tools/FilterTestTool/FilterTest.py", line 91, in __init__
    self.init_plot(log_name)
  File "/mnt/i/github/ardupilot-master/Tools/FilterTestTool/FilterTest.py", line 416, in init_plot
    (self.acc_filtered_fft_ax_x, self.acc_filtered_fft_ax_y, self.acc_filtered_fft_ax_z))
  File "/mnt/i/github/ardupilot-master/Tools/FilterTestTool/FilterTest.py", line 333, in create_controls
    filters_key, time_list, sample_lists, signal_shapes, fft_shapes, filt_color)
  File "/mnt/i/github/ardupilot-master/Tools/FilterTestTool/FilterTest.py", line 307, in create_filter_control
    signal_shape=signal_shapes, fft_shape=fft_shapes:
  File "/mnt/i/github/ardupilot-master/Tools/FilterTestTool/FilterTest.py", line 204, in create_slider
    slider = Slider(ax_slider, name, 0, max, valinit=np.sqrt(max * value), valstep=1, color=color)
  File "/usr/lib/python3/dist-packages/matplotlib/widgets.py", line 342, in __init__
    self.poly = ax.axvspan(valmin, valinit, 0, 1, **kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/axes/_axes.py", line 853, in axvspan
    p = mpatches.Polygon(verts, **kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/patches.py", line 921, in __init__
    Patch.__init__(self, **kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/patches.py", line 111, in __init__
    self.update(kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/artist.py", line 902, in update
    for k, v in props.items()]
  File "/usr/lib/python3/dist-packages/matplotlib/artist.py", line 902, in <listcomp>
    for k, v in props.items()]
  File "/usr/lib/python3/dist-packages/matplotlib/artist.py", line 895, in _update_property
    raise AttributeError('Unknown property %s' % k)
AttributeError: Unknown property valstep

**Comment by andyp1per** [comment]
Never mind, seems an old version of matlibplot - working my way down the rabbit hole!

**Comment by guglie** [comment]
> Never mind, seems an old version of matlibplot - working my way down the rabbit hole!

Yes, I've used it with matplotlib 3.0.2

**Pull request merged**